lib/scrub/scrub_runner.rb

```diff
@@ -0,0 +1,111 @@
+# A fully automated scrubber would:
+# Check member directories for new member-submitted files
+# Compare remote files to processed files
+# Copy new files to work directory
+# Validate and scrub files into loadable format
+# Load files into db
+# Upload report to member dir
+# ... should be able to do this for any number of members/files, at a whim or by cron,
+# ideally using a queue or pool of workers.
+
+require "data_sources/ht_organizations"
+require "loader/file_loader"
+require "loader/holding_loader"
+require "scrub/autoscrub"
+require "fileutils"
+
+# Example:
+# runner = Scrub::ScrubRunner.new
+# runner.run_all_members
+# runner.run_some_members(["umich", ..., "harvard"])
+# runner.run_one_member("umich")
+
+module Scrub
+  # Scrubs and loads any new member-uploaded files.
+  class ScrubRunner
+    def initialize
+      @members = DataSources::HTOrganizations.new.members.keys.sort
+      if Settings.scrubbed_files_dir.nil?
+        raise "Need Settings.scrubbed_files_dir to be set"
+      end
+    end
+    # see lib/reports/etas_organization_overlap_report.rb for rclone stuff,
+    # should be broken out into its own class?
+
+    def run_all_members
+      puts "Run all members."
+      run_some_members(@members)
+    end
+
+    def run_some_members(members)
+      puts "Run members: #{members}."
+      members.each do |member|
+        run_one_member(member)
+      end
+    end
+
+    def run_one_member(member)
+      puts "Running member #{member}."
+      new_files = check_new_files(member)
+      puts "Found #{new_files.size} new files: #{new_files.join(", ")}."
+      new_files.each do |new_file|
+        run_file(member, file)
+      end
+    end
+
+    # This should ideally spin off another (set of?) worker thread(s).
+    def run_file(member, file)
+      puts "running file #{file} for member #{member}"
+      download_to_work_dir(member, new_file)
+      scrubber = Scrub::Autoscrub.new(new_file)
+      scrubber.run
+      # figure out the output files that was just generated
+      # todo: implement scrubber.reports and/or do it with Scrub::ScrubOutputStructure
+      upload_to_member(member, scrubber.reports.latest)
+      holding_loader = Loader::HoldingLoader.for(scrubber.output.latest)
+      Loader::FileLoader.new(holding_loader).load(scrubber.output.latest)
+    end
+
+    # Check member-uploaded files for any not previously seen files
+    def check_new_files(member)
+      puts "check new files for member #{member}"
+      remote_files = [] # todo: something something rclone
```

Comment:
Should implement a class for rclone that implements the functionality we need (upload, download, list files)

Can use something like https://github.com/hathitrust/holdings-backend/blob/main/spec/reports/etas_organization_overlap_report_spec.rb#L8-L19 to use a local path for test purposes rather than using dropbox

Reply:
Is a file showing up in dropbox sufficient cause for starting a scrub & load job? @mwarin says yes

Reply:
A problem is that members can make their own directories inside their own dropbox. We can't control what they name things. Anything that doesn't match should probably trigger a warning.

Information from file name:
- member id - if on dropbox, should be determinable from the folder, but doesn't hurt to add it (since we have to add a file naming scheme)
- update type (full or partial) which we aren't using yet
- mono/multi/serial which we may not use
- date string - Dropbox has proved unreliable with file timestamping, so we should ask for this
- whether it's compressed - can only work with uncompressed or .gz

Reply:
What happens if somebody renames a file? More generally, how do we want to track where things are in the process?

Reply:
`Utils::FileTransfer` in combination with `DataSources::DirectoryLocator` are pretty much doing this now.